plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -447,10 +456,16 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
     fun tryResumeJob() = projectCoroutineScope(project).launch {
         try {
             val notYetResumed = isResumingJob.compareAndSet(false, true)
+            // If the job is already running, compareAndSet will return false because the expected
+            // behavior is that the job is not running when trying to resume
             if (!notYetResumed) {
                 return@launch
             }
 
+            // Gather project details
+            val validationResult = validate(project)
+            sendValidationResultTelemetry(validationResult, true)
```

Comment:
activeConnectionChanged also calls tryResumeJob. Will that cause multiple events being emitted?

Reply:
In that case yes it will. We can:
1. Leave it since its possible their project changed post re-connecting
2. Move it to the parent function

Reply:
Either of these seem fine just let us know what you choose